### DISCUSSION TITLE
Feature Request: Parquet Union By Name param only reads files/columns required for query

### DISCUSSION THREAD

**Discussion by danthegoodman1** [duckdb/duckdb#5809] [Ideas]
Related to https://github.com/duckdb/duckdb/pull/5716

With the new `union_by_name` we lose the benefit of only reading the columns that are required for the select/where clauses, but rather read in the entire file ([according to this comment](https://github.com/duckdb/duckdb/pull/5716#issuecomment-1357204325)). Furthermore it reads the entire file regardless of whether it contains the column or not, so there is an opportunity to avoid reading files all together.

Ideally this feature can be extended so that only the relevant columns are read which can greatly reduce the data pulled and query processing times.

**Comment by danthegoodman1** [comment]
@douenergy maybe you can confirm that

> we lose the benefit of only reading the columns that are required for the select/where clauses

Is true? I am assuming that based on

> it reads the entire file regardless of whether it contains the column or not

**Reply by douenergy** [reply]
It's True.

```sql
CREATE OR REPLACE TABLE t1 AS  SELECT a , a+5 as b FROM generate_series(1,10) as g(a);
CREATE OR REPLACE TABLE t2 AS  SELECT a+3 as a, a+33 as c  FROM generate_series(1,10) as g(a);

COPY t1 TO 'p1.parquet' (FORMAT PARQUET);
COPY t2 TO 'p2.parquet' (FORMAT PARQUET);
```

```sql
SELECT * 
FROM parquet_scan('p*.parquet', union_by_name=true) 
WHERE b is not null 
```

This query currently dont `ApplyFilter` in  https://github.com/duckdb/duckdb/pull/5716

**Reply by danthegoodman1** [reply]
Thank you for confirming!
